lib/instrumentation/modules/http2.js

```diff
@@ -127,4 +131,38 @@ module.exports = function (http2, agent) {
     }
     return result
   }
+
+  function wrapConnect (orig) {
+    return function (host) {
+      const ret = orig.apply(this, arguments)
+      shimmer.wrap(ret, 'request', orig => wrapRequest(orig, host))
+      return ret
+    }
+  }
+
+  function wrapRequest (orig, host) {
+    return function (headers) {
+      var span = agent.buildSpan()
+      var id = span && span.transaction.id
+
+      agent.logger.debug('intercepted call to http2.request %o', { id })
+
+      var req = orig.apply(this, arguments)
+      if (!span) return req
+
+      var protocol = req.agent && req.agent.protocol
+      agent.logger.debug('request details: %o', { protocol, host, id })
+
+      var method = headers[':method'] || 'GET'
```

Comment:
I'm not sure, but I figured there's no harm in trying to be safe here.

Reply:
Did you consider instead falling back to something like `UNKNOWN` or something similar indicating that the method wasn't known... or maybe the default method is just `GET`?

Reply:
So `:method` is not specifically included unless the client explicitly includes it in the input objects. It doesn't seem to actually be required. I'm not sure if that's a bug in Node.js or if it's supposed to work like that. 🤔

Reply:
Found this in the [HTTP2 spec - 8.1.2.3 Request Pseudo-Header Fields](https://http2.github.io/http2-spec/#rfc.section.8.1.2.3):

> All HTTP/2 requests MUST include exactly one valid value for the `:method`, `:scheme`, and `:path` pseudo-header fields, unless it is a CONNECT request ([Section 8.3](https://http2.github.io/http2-spec/#CONNECT)). An HTTP request that omits mandatory pseudo-header fields is malformed ([Section 8.1.2.6](https://http2.github.io/http2-spec/#malformed)).

I mean, it's always good to be extra safe, but 1) it sounds like a bug in the Node.js http2 implementation, and 2) if we need to guard against a missing `:method` header, we should maybe also guard against a missing `:path` header right?

But lets first be sure it actually can happen, then if it can, we should still consider if we want the `:method` fallback as it will not throw if it's not there. It will just use the string `undefined` which might actually be more clear. But `url.parse(undefined)` on the other hand will throw, so if that's possible we need to guard against `:path` being `undefined`.

Reply:
Interesting...I'll look at that some more. 🤔

Reply:
So it appears to be added on the client end. https://github.com/nodejs/node/blob/master/lib/internal/http2/core.js#L1364-L1383

So I can just assume a blank `:path` is `/` and a blank `:method` is `GET`. I don't see a way to identify the `:scheme`, in the event it is absent, however we're not currently using that.